src/elements/fields/TextField.jsx

```diff
@@ -75,6 +85,86 @@ function getTextFieldProps(servar, styles, value) {
     }
 }
 
+function getFieldMaskMeta(fieldMask) {
+    if (fieldMask === '' || fieldMask === undefined || fieldMask === null)
+        return {
+            fieldMaskRegex: '',
+            fieldMaskComparisionString: '',
+            fieldMaskString: '',
+            deterministicPattern: true
+        };
+
+    const [
+        fieldMaskRegex,
+        fieldMaskComparisionString,
+        fieldMaskString,
+        deterministicPattern
+    ] = generateRegexString(fieldMask);
+
+    return {
+        fieldMaskRegex,
+        fieldMaskComparisionString,
+        fieldMaskString,
+        deterministicPattern
+    };
+}
+
+const validateNewFieldValue = (
```

Comment:
what about copy-pasting, or if the user deletes/inserts/pastes a replacement for a section of text?

Reply:
Copy-pasting has been implemented.